docs/mp/config/02_MP_config_sources.adoc

```diff
@@ -0,0 +1,155 @@
+///////////////////////////////////////////////////////////////////////////////
+
+    Copyright (c) 2020 Oracle and/or its affiliates.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+///////////////////////////////////////////////////////////////////////////////
+
+= Microprofile Config Sources
+:pagename: MicroProfileSources
+:description: MicroProfile Config Sources
+:keywords: helidon, mp, ordinal, mpconfig, yamlmpconfig
+
+
+A Config Source provides configuration values from different sources such as property files and user classes that are registered by the application.
+
+== Understanding the Ordering of Default Config Sources
+
+The default MicroProfile Config Sources are:
+
+* System properties (ordinal=400)
+* Environment variables (ordinal=300)
+* /META-INF/microprofile-config.properties (ordinal=100)
+
+Each Config Source has a ordinal that determines the priority of the Config Source. A Config Source with higher ordinal has high priority as compared to the Config Source with lower ordinal. The values taken from the high-priority Configsource overrides the values from low-priority Config Source. This helps to customize the configuration of Config Sources using external Config Source if an external Config Source has higher ordinal values than the built-in Config Sources of the application.
+
+
+== Creating Custom Config Sources
+
+Custom Config Sources are loaded using the Java ServiceLoader pattern and can be registered by providing a file named `org.eclipse.microprofile.config.spi.ConfigSource` in the `/META-INF/services/` folder.
+
+You need to use the following methods and implement the interface `org.eclipse.microprofile.config.spi.ConfigSource` to create a custom Config Source:
+
+* `String getName()`
+* `Map<String, String> getProperties()`
+* `String getValue(String key)`
+* `getOrdinal()`
+
+=== Example of a Custom Config Source
+
+[source,java]
+----
+public class CustomConfigSource implements ConfigSource {

```

Comment:
I think the fields used to be constants - please change it back to constants, or rename the fields and reformat the source code.
If a fields is in all capitals, it is usually a constant (and must be `static final` - in this case `private static final`)
Fields that are not constants use camel case with initial lower case (so it would be `name`, `priority` and `properties`)

In this example, the constant is expected.